### PR TITLE
Set write-cache to off explicitly

### PR DESCRIPTION
**Pull request by derekbit**
Longhorn/longhorn#5382

Signed-off-by: Derek Su <derek.su@suse.com>

### CODE REVIEW

File: iscsidev/iscsi.go [hunk 1]
```diff
@@ -115,6 +115,9 @@ func (dev *Device) CreateTarget() (err error) {
 	if err := iscsi.SetLunThinProvisioning(dev.targetID, TargetLunID); err != nil {
 		return err
 	}
+	if err := iscsi.DisableWriteCache(dev.targetID, TargetLunID); err != nil {
```
Comment (shuo-wu) [inline iscsidev/iscsi.go:118]: NIT: Add a simple comment explaining why it disables  the write cache

**Review by shuo-wu — DISMISSED** [review]
LGTM

File: iscsi/target.go [hunk 2]
```diff
@@ -118,6 +118,11 @@ func SetLunThinProvisioning(tid int, lun int) error {
 	return UpdateLun(tid, lun, map[string]string{"thin_provisioning": "1"})
 }
 
+// DisableWriteCache will set param write-cache to false for the LUN
+func DisableWriteCache(tid int, lun int) error {
```
Comment (joshimoo) [inline iscsi/target.go:122]: NIT: can you add a link to the reference of the mode page parameters, otherwise the below is just magic :)

**Review by joshimoo — APPROVED** [review]
LGTM

File: iscsi/target.go (hunk 2 above)
Comment (derekbit) [reply iscsi/target.go:122]: Added. Thank you.

File: iscsidev/iscsi.go (hunk 1 above)
Comment (derekbit) [reply iscsidev/iscsi.go:121]: Added. Thank you.

**Review by joshimoo — APPROVED** [review]

**Comment by derekbit** [comment]
@Mergifyio backport v1.3.x

**Pull request merged**